traffic_portal/app/src/modules/private/cdns/config/ConfigController.js

```diff
@@ -215,16 +158,34 @@ var ConfigController = function(cdn, currentSnapshot, newSnapshot, $scope, $stat
 	$scope.navigateToPath = locationUtils.navigateToPath;
 
 	angular.element(document).ready(function () {
-		$scope.diffConfig(0);
-		$scope.diffContentRouters(0);
-		$scope.diffMonitors(0);
-		$scope.diffContentServers(0);
-		$scope.diffDeliveryServices(0);
-		$scope.diffEdgeLocations(0);
-		$scope.diffStats(0);
+
+		$('table.changes').dataTable({
```

Comment:
`s/\$\(([^\)]+)\)/document\.querySelectoryAll\(\1\)[0]/g`

pls let jQuery die

Reply:
they are called "jquery datatables" afterall. :)

what is this supposed to do?

`s/\$\(([^\)]+)\)/document\.querySelectorAll\(\1\)\[0\]/g`

Reply:
Since Angular comes with JQuery (sort of) I was hoping you could do this with just `angular.element(document.QuerySelector('table.changes')).dataTable({...});` but actually it seems like their "JQuery-Lite" may not provide `dataTable` properties. In general it's advised that AngularJS projects avoid doing DOM manipulation outside of Directives - including using JQuery in Controllers (it messes with Angular's "digest" cycle because DOM manipulation events that happen outside of Directives cannot be captured by Angular). So whenever I see it I'm wary of wether or not it's actually necessary.

Can't find a lot of these options in the [datatables docs](https://www.datatables.net/reference/option/) though. What does pre-pending those lower-case letters do?

Reply:
yeah, you're probably right and there are just a few instances (actually all of the dataTable instances where the datatable is initialized/configured) i believe where TP uses jquery and it's probably because not using jquery didn't work and i didn't want to bother spending too much time trying to figure out why. Im not sure what you're asking here:

> What does pre-pending those lower-case letters do?

Reply:
There's a [`lengthMenu` option](https://datatables.net/reference/option/lengthMenu), but not an `aLengthMenu` option, there's no `iDisplayLength`/`displayLength` option anywhere that I can see, there's no `aaSorting`/`sorting` option, but there is an `ordering` option.

Reply:
good question. most of these settings were set originally via a stackoverflow search i'm sure and they've simply propagated via copy/paste...

Reply:
I actually can't even find where datatables is included as a TP dependency. Unless it's bundled with `angular-bootstrap` (which doesn't have a readme, and [the page on npm](https://www.npmjs.com/package/angular-bootstrap) seems to indicate that exactly one version exists - 0.12.2 from 4 years ago, but we explicitly require 0.14.3 somehow) or `jquery` for some insane reason.

Reply:
there are a handful of javascript libraries that are vendored for a lack of a better word and those can be found in index.html. looks like jquery datatables 1.10.16 is being used.

Reply:
According to the docs for [`order`](https://datatables.net/reference/option/order) and [`lengthMenu`](https://datatables.net/reference/option/lengthMenu) they've both been available since 1.10. 

Looking at the value of `iDisplayLength`, I think you're using it to set the initial length of the display (in rows). According to [the docs for `pageLength`](https://datatables.net/reference/option/pageLength) that does the same thing and has been available since 1.10.

I think it'd be better to use documented options - undocumented stuff is hard to maintain.